pkg/domain/infra/tunnel/images.go

```diff
@@ -347,6 +348,14 @@ func (ir *ImageEngine) Config(_ context.Context) (*config.Config, error) {
 }
 
 func (ir *ImageEngine) Build(_ context.Context, containerFiles []string, opts entities.BuildOptions) (*entities.BuildReport, error) {
+	_, err := os.Stat(opts.ContextDirectory)
+	if err != nil { // we are remote and the context dir does not exist but it did on the client
+		err = os.Mkdir(filepath.Join(".", opts.ContextDirectory), os.ModeDir)
+		if err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
This will still run on the client side so it does not fix your problem. Also I don't think creating a directory will fix the root problem here which seems to be that the Containerfile is not inside the context dir.

Reply:
if that is the issue I can just make sure it is enforced

Reply:
If it works local it should work with remote too. The bindings already add the containerfile to the tar so it should be on the server somewhere. https://github.com/containers/podman/blob/fa33f3527f14edbf7407ba3ba482ef839debfb39/pkg/bindings/images/build.go#L382

I think the server is just looking at the wrong place for the file.

Reply:
ok @jwhonce @Luap99 thanks for the help on narrowing down the issue I think I can probably debug this now to figure out what's going on. The only thing that doesn't make sense to me is why we allow a context dir that the containerfile is not in? The issue #13293 seems to be failing due to the context dir not existing so that must be getting skipped in the tar (maybe when empty?) either way I think this kind of opens a discussion on whether we want to fix this by 1) not allowing a random context dir that the containerfile is not in or 2) making sure we tar the contextDir.